app/multilistener.go

```diff
@@ -0,0 +1,74 @@
+package app
+
+import (
+	"context"
+	"net"
+
+	"github.com/pkg/errors"
+	"github.com/target/goalert/util/log"
+)
+
+type multiListener struct {
+	listeners []net.Listener
+
+	ch    chan net.Conn
+	errCh chan error
+}
+
+func newMultiListener(ln ...net.Listener) *multiListener {
+	ml := multiListener{listeners: ln}
+	ml.ch = make(chan net.Conn)
+	ml.errCh = make(chan error)
+
+	for _, l := range ln {
+		go ml.listen(l)
+	}
+	return &ml
+}
+
+// listen waits for and returns the next connection for the listener.
+func (ml multiListener) listen(l net.Listener) {
+	for {
+		c, err := l.Accept()
+		if err != nil {
+			ml.errCh <- err
+			return
+		}
+		ml.ch <- c
```

Comment:
This could hang if we accept a connection right before `Close` is called and `Accept` is never called again.

Probably needs to be a `select` that listens for some sort of `shutdown` channel, and then calls `Close` on `c` if we are shutting down the listener. Otherwise, we could get a stalled goroutine and a hung connection.